DUI3-DX/Converters/Autocad/Speckle.Converters.AutocadShared/Extensions/ListExtensions.cs

```diff
@@ -14,4 +14,44 @@ public static SOG.Polyline ConvertToSpecklePolyline(this List<double> pointList,
 
     return new(pointList, speckleUnits);
   }
+
+  public static List<AG.Point2d> ConvertToPoint2d(this List<double> pointList, double conversionFactor = 1)
+  {
+    // throw if list is malformed
+    if (pointList.Count % 2 != 0)
+    {
+      throw new System.ArgumentException("Point list of xy values is malformed.");
+    }
+
+    List<AG.Point2d> points2d = new();
+    for (int i = 1; i < pointList.Count; i += 2)
+    {
+      points2d.Add(new AG.Point2d(pointList[i - 1] * conversionFactor, pointList[i] * conversionFactor));
+    }
+
+    return points2d;
+  }
+
+  public static List<AG.Point3d> ConvertToPoint3d(this List<double> pointList, double conversionFactor = 1)
+  {
+    // throw if list is malformed
+    if (pointList.Count % 3 != 0)
+    {
+      throw new System.ArgumentException("Point list of xyz values is malformed.");
+    }
+
+    List<AG.Point3d> points3d = new();
```

Comment:
I would prefer we presize these lists if they will be of a known size, and have potential to be really really large.
This is for performance reasons (since the list never needs to be dynamically resized)

```cs
List<Ag.Point3d point3d = new(pointList.Count * 3);
```

I'll try and do a round of profiling after the alpha is shipped, and we can identify and fix bottlenecks then if you'd prefer to merge this PR with out this change

Reply:
Good point Jedd, doing it.

Reply:
But I think it is `pointList.Count / 3` not multiple

Reply:
Yep you're absolutely right, good catch